src/tests/token/erc721/test_erc721.cairo

```diff
@@ -850,159 +909,175 @@ fn test_safeTransferFrom_to_owner_camel() {
 #[test]
 fn test_safe_transfer_from_approved() {
     let mut state = setup();
+    let contract_address = test_address();
     let receiver = setup_receiver();
     let token_id = TOKEN_ID;
     let owner = OWNER();
 
     assert_state_before_transfer(owner, receiver, token_id);
 
-    testing::set_caller_address(owner);
+    start_cheat_caller_address(contract_address, owner);
     state.approve(OPERATOR(), token_id);
-    utils::drop_event(ZERO());
 
-    testing::set_caller_address(OPERATOR());
+    let mut spy = spy_events();
+
+    start_cheat_caller_address(contract_address, OPERATOR());
     state.safe_transfer_from(owner, receiver, token_id, DATA(true));
-    assert_only_event_transfer(ZERO(), owner, receiver, token_id);
+    spy.assert_only_event_transfer(contract_address, owner, receiver, token_id);
 
     assert_state_after_transfer(owner, receiver, token_id);
 }
 
 #[test]
 fn test_safeTransferFrom_approved() {
     let mut state = setup();
+    let contract_address = test_address();
     let receiver = setup_receiver();
     let token_id = TOKEN_ID;
     let owner = OWNER();
 
     assert_state_before_transfer(owner, receiver, token_id);
 
-    testing::set_caller_address(owner);
+    start_cheat_caller_address(contract_address, owner);
     state.approve(OPERATOR(), token_id);
-    utils::drop_event(ZERO());
 
-    testing::set_caller_address(OPERATOR());
+    let mut spy = spy_events();
+
+    start_cheat_caller_address(contract_address, OPERATOR());
     state.safeTransferFrom(owner, receiver, token_id, DATA(true));
-    assert_only_event_transfer(ZERO(), owner, receiver, token_id);
+    spy.assert_only_event_transfer(contract_address, owner, receiver, token_id);
 
     assert_state_after_transfer(owner, receiver, token_id);
 }
 
 #[test]
+#[ignore]
 fn test_safe_transfer_from_approved_camel() {
     let mut state = setup();
+    let contract_address = test_address();
+    let mut spy = spy_events();
     let receiver = setup_camel_receiver();
     let token_id = TOKEN_ID;
     let owner = OWNER();
 
     assert_state_before_transfer(owner, receiver, token_id);
 
-    testing::set_caller_address(owner);
+    start_cheat_caller_address(contract_address, owner);
     state.approve(OPERATOR(), token_id);
-    utils::drop_event(ZERO());
 
-    testing::set_caller_address(OPERATOR());
+    start_cheat_caller_address(contract_address, OPERATOR());
     state.safe_transfer_from(owner, receiver, token_id, DATA(true));
-    assert_only_event_transfer(ZERO(), owner, receiver, token_id);
+    spy.assert_only_event_transfer(contract_address, owner, receiver, token_id);
 
     assert_state_after_transfer(owner, receiver, token_id);
 }
 
 #[test]
+#[ignore]
 fn test_safeTransferFrom_approved_camel() {
     let mut state = setup();
+    let contract_address = test_address();
+    let mut spy = spy_events();
```

Comment:
Despite this test is ignored, I think we should move spy initialization further in the function (after the `approve` call)